### PR TITLE
Prepare the v3.5.0 release

### PR DESCRIPTION
**Pull request by 4383**
Depends-on #324

### CODE REVIEW

**Comment by 4383** [comment]
It could be worth to cut this release and land these fixes before the next major release where the support of python 2.7 will be dropped.

File: ChangeLog.rst [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+New in version 3.5.0
+--------------------
+* Fixing a socket closing on ``MemcacheUnexpectedCloseError``
```
Comment (jparise) [inline ChangeLog.rst:6]:
```suggestion
* Sockets are now closed on ``MemcacheUnexpectedCloseError``.
```

File: ChangeLog.rst [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+* Added support of TCP keepalive for client sockets on Linux platforms
```
Comment (jparise) [inline ChangeLog.rst:7]:
```suggestion
* Added support for TCP keepalive for client sockets on Linux platforms.
```

File: ChangeLog.rst [hunk 3]
```diff
@@ -1,6 +1,12 @@
...
+* Added retrying mechanisms by wrapping clients
```
Comment (jparise) [inline ChangeLog.rst:8]: This hasn't been merged yet.

Comment (4383) [reply ChangeLog.rst:8]: Yes I know this is why I added a depends-on warning in the description of this pull request

File: ChangeLog.rst [hunk 4]
```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+New in version 3.5.0
+--------------------
+* Sockets are now closed on ``MemcacheUnexpectedCloseError``.
+* Added support for TCP keepalive for client sockets on Linux platforms.
+* Added retrying mechanisms by wrapping clients.
```
Comment (4383) [inline ChangeLog.rst:8]: Waiting for #324

Comment (martinnj) [reply ChangeLog.rst:8]: This was just merged. 👍

Comment (4383) [reply ChangeLog.rst:8]: So we are ready to go

**Review by jparise — APPROVED** [review]

**Pull request merged**